### ISSUE TITLE
Broken validation of JS files in Flow codebase

### ISSUE THREAD

**Issue by jurosh** [flowtype/flow-for-vscode#250]
As per closed issue https://github.com/flowtype/flow-for-vscode/issues/13

There was discussion about issues with Flow complaining about JS validation rules - not knowing types. And it was closed with reason that you can turn off validation for ALL js files.

But that's just wrong as we lose validation in JS files where is no `// @flow` statement on top of file and we can have there many errors unnoticed.

We might need something like (following config is not functional):

```
"[flow]": {
    "javascript.validate.enable": false
}
```

**Comment by betaorbust** [comment]
Yes please. Using the global switch on validation means that working in a codebase without complete flow coverage is a _huge_ pain/not really viable.
I like VSCode, I like flow, but requiring that a file be `// @flow` enabled or else you lose huge chunks of editor functionality is... not kind.

**Comment by kuprik** [comment]
hey guys, I just started using JS for my first web project. I pulled some sources from react-grid-layout and it seems to use flow. Most of my project are react jsx code, and not flow. Immediately I find that if built-in JS validation is enabled in VS Code (default), flow files complain. Disabling validation globally is bad for all other sources. I am curious this issue never appeared till present on Gihub (and I just literally started my web dev career 2 weeks ago and immediately hit this)
